Assets/Guns/Scripts/GunRecoil.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+using System.Collections;
+
+public class GunRecoil : MonoBehaviour
+{
+    Vector3 newPos;
+    Vector3 startPos;
+
+    public float moveBack = .01f;
+    public float speed = 10f;
+
+    private void Start()
+    {
+        newPos = startPos = transform.localPosition;
+    }
+
+    private void Update()
+    {
+        transform.localPosition = Vector3.Slerp(transform.localPosition, newPos, speed * Time.deltaTime);
+
+        if (Input.GetMouseButtonDown(0))
```

Comment:
We are using the new input system, if this is debugging code, please remove before merging with main